zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,317 @@
+
+        # 参数验证
+        if up.slice_offset >= up.total_size:
+            raise Exception('invalid slice offset header: %s, total_size: %d' %
+                            (up.slice_offset, up.total_size))
+
+        return up
+
+    def get_upload_task(self, param):
+        task = self.upload_tasks.get_task(param.image_uuid)
+        if task is None:
+            raise Exception('image not found %s' % param.image_uuid)
```

Ruff (0.13.1)

Create your own exception

(TRY002)